receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -167,6 +176,10 @@ func (s *scraper) getProcessMetadata() ([]*processMetadata, error) {
 
 	var errs scrapererror.ScrapeErrors
 
+	if err := s.refreshHandleCounts(); err != nil {
```

Comment:
I might be misunderstanding what `getProcessHandles` does, but isn't this refresh still too late? It seems like `handles` would always be empty on the first invocation of `getProcessMetadata`.

Reply:
Ah thanks, I got mixed up between handles and handle counts there.